common/tasks/interleaved_weighted_round_robin.go

```diff
@@ -254,18 +185,10 @@ func (s *InterleavedWeightedRoundRobinScheduler[T, K]) TrySubmit(
 }
 
 func (s *InterleavedWeightedRoundRobinScheduler[T, K]) eventLoop() {
-	checkRateLimiterEnabledTimer := time.NewTicker(checkRateLimiterEnabledInterval)
-	defer checkRateLimiterEnabledTimer.Stop()
-
 	for {
 		select {
 		case <-s.notifyChan:
 			s.dispatchTasksWithWeight()
-		case <-checkRateLimiterEnabledTimer.C:
-			s.rateLimiterConfig.Store(rateLimiterConfig{
-				enableThrottle:   s.options.EnableRateLimiter(),
-				enableShadowMode: s.options.EnableRateLimiterShadowMode(),
```

Comment:
Are we removing the ability to dynamically turn shadow mode on/off without a restart?

Reply:
yeah. In the new code the config need to be checked per task on TrySubmit() if we want the truly dynamic behavior and it will be in the hot path. I personally think that's unnecessary:
- once enabled it's unlikely we will turn if off. 
- the actual rate is still truly dynamic and by using a high enough number, the rate limiter is effectively disabled. That's actually why in the initial version I remove the feature flag for enabling rate limiter entirely. (Added back for some other reasons)
- Restart for enabling a feature is safer than rolling out a dc.